### ISSUE TITLE
OCA\\GroupFolders\\Folder\\FolderManager::getFolder(): Return value must be of type array, bool returned

### ISSUE THREAD

**Issue by solracsf** [nextcloud/groupfolders#2241]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

```
{
  "reqId": "4tSQzuDot3cA1vx7Q8go",
  "level": 3,
  "time": "2023-01-26T12:02:00+01:00",
  "user": "admin",
  "app": "index",
  "method": "GET",
  "url": "/apps/groupfolders/folders/undefined",
  "message": "OCA\\GroupFolders\\Folder\\FolderManager::getFolder(): Return value must be of type array, bool returned in file '/var/www/nextcloud/apps/groupfolders/lib/Folder/FolderManager.php' line 246",
  "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36",
  "version": "24.0.9.2",
  "exception": {
    "Exception": "Exception",
    "Message": "OCA\\GroupFolders\\Folder\\FolderManager::getFolder(): Return value must be of type array, bool returned in file '/var/www/nextcloud/apps/groupfolders/lib/Folder/FolderManager.php' line 246",
    "Code": 0,
    "Trace": [
      {
        "file": "/var/www/nextcloud/lib/private/AppFramework/App.php",
        "line": 172,
        "function": "dispatch",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "->",
        "args": [
          {
            "__class__": "OCA\\GroupFolders\\Controller\\FolderController"
          },
          "getFolder"
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/private/Route/Router.php",
        "line": 298,
        "function": "main",
        "class": "OC\\AppFramework\\App",
        "type": "::",
        "args": [
          "OCA\\GroupFolders\\Controller\\FolderController",
          "getFolder",
          {
            "__class__": "OC\\AppFramework\\DependencyInjection\\DIContainer"
          },
          [
            "undefined",
            "groupfolders.Folder.getFolder"
          ]
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/base.php",
        "line": 1030,
        "function": "match",
        "class": "OC\\Route\\Router",
        "type": "->",
        "args": [
          "/apps/groupfolders/folders/undefined"
        ]
      },
      {
        "file": "/var/www/nextcloud/index.php",
        "line": 36,
        "function": "handleRequest",
        "class": "OC",
        "type": "::",
        "args": []
      }
    ],
    "File": "/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
    "Line": 165,
    "Previous": {
      "Exception": "TypeError",
      "Message": "OCA\\GroupFolders\\Folder\\FolderManager::getFolder(): Return value must be of type array, bool returned",
      "Code": 0,
      "Trace": [
        {
          "file": "/var/www/nextcloud/apps/groupfolders/lib/Controller/FolderController.php",
          "line": 72,
          "function": "getFolder",
          "class": "OCA\\GroupFolders\\Folder\\FolderManager",
          "type": "->",
          "args": [
            0,
            2
          ]
        },
        {
          "file": "/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
          "line": 225,
          "function": "getFolder",
          "class": "OCA\\GroupFolders\\Controller\\FolderController",
          "type": "->",
          "args": [
            0
          ]
        },
        {
          "file": "/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
          "line": 133,
          "function": "executeController",
          "class": "OC\\AppFramework\\Http\\Dispatcher",
          "type": "->",
          "args": [
            {
              "__class__": "OCA\\GroupFolders\\Controller\\FolderController"
            },
            "getFolder"
          ]
        },
        {
          "file": "/var/www/nextcloud/lib/private/AppFramework/App.php",
          "line": 172,
          "function": "dispatch",
          "class": "OC\\AppFramework\\Http\\Dispatcher",
          "type": "->",
          "args": [
            {
              "__class__": "OCA\\GroupFolders\\Controller\\FolderController"
            },
            "getFolder"
          ]
        },
        {
          "file": "/var/www/nextcloud/lib/private/Route/Router.php",
          "line": 298,
          "function": "main",
          "class": "OC\\AppFramework\\App",
          "type": "::",
          "args": [
            "OCA\\GroupFolders\\Controller\\FolderController",
            "getFolder",
            {
              "__class__": "OC\\AppFramework\\DependencyInjection\\DIContainer"
            },
            [
              "undefined",
              "groupfolders.Folder.getFolder"
            ]
          ]
        },
        {
          "file": "/var/www/nextcloud/lib/base.php",
          "line": 1030,
          "function": "match",
          "class": "OC\\Route\\Router",
          "type": "->",
          "args": [
            "/apps/groupfolders/folders/undefined"
          ]
        },
        {
          "file": "/var/www/nextcloud/index.php",
          "line": 36,
          "function": "handleRequest",
          "class": "OC",
          "type": "::",
          "args": []
        }
      ],
      "File": "/var/www/nextcloud/apps/groupfolders/lib/Folder/FolderManager.php",
      "Line": 246
    },
    "CustomMessage": "--"
  },
  "id": "63d3b30ad1446"
}
```

### Expected behaviour
Tell us what should happen

### Actual behaviour
Tell us what happens instead

### Server configuration

**PHP version:** 8.1

**Nextcloud version:** (see Nextcloud admin page) 24.0.9

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** Saml

**Comment by rawtaz** [comment]
Is this just for your own reference? Otherwise, can you share how to reproduce this problem? It might help others know what to look out for, avoid or be aware of, or even not do :-)

**Comment by solracsf** [comment]
> Otherwise, can you share how to reproduce this problem?

Well, if i knew hot to reprodce it I would have write it 😄 
I have just a bunch of these in the logs.

**Comment by donquixote** [comment]
It happens if you request a non-existing group folder.
E.g. GET http://nextcloud/apps/groupfolders/folder/999

The bug is quite obvious in the v13.1.1 version of FolderManager.
https://github.com/nextcloud/groupfolders/blob/v13.1.1/lib/Folder/FolderManager.php
```php
	public function getFolder(int $id, int $rootStorageId): array {
		[...]
		return $row ? [
			[...]
		] : false;
	}
```
Obviously this will break if it attempts to return false.


It is already fixed in master in https://github.com/nextcloud/groupfolders/commit/6e3d73e5d1aeec7e9ffe8e1cf88743cad41d5609#diff-56ea4cc1339b392e00771c8f47ef11e3f4c33a3a7cc0c11ebc92b766a760e565L268, by removing the return type.

```diff
-	public function getFolder(int $id, int $rootStorageId): array {
+	public function getFolder(int $id, int $rootStorageId) {
```

**Comment by donquixote** [comment]
The question is, can we expect a release soon that would contain this fix, or should we do a hotfix?

**Comment by solracsf** [comment]
Fixed by v14.0.0
